ext/js/pages/settings/languages-controller.js

```diff
@@ -25,17 +25,25 @@ export class LanguagesController {
     constructor(settingsController) {
         /** @type {import('./settings-controller.js').SettingsController} */
         this._settingsController = settingsController;
+        /** @type {string} */
+        this._lastSelectedLanguage = '';
     }
 
     /** */
     async prepare() {
         const languages = await this._settingsController.application.api.getLanguageSummaries();
         languages.sort((a, b) => a.name.localeCompare(b.name, 'en'));
-        this._fillSelect(languages);
+        const languageSelect = this._fillSelect(languages);
+        languageSelect.addEventListener(
```

Comment:
This is actually kind of brittle. To prompt users whether they'd like to update the current profile settings upon changing the language selection dropdown is falsely triggered by the following scenarios:

- When I change profiles which update the language selected
- When I load the page and the `prepare()` statement runs, changing the value (this is guarded via a clause)

If someone knows how to guard against the first condition OR know a better place to hook this logic, that would be great